backend/src/nodes/onnx_nodes.py

```diff
@@ -47,18 +48,32 @@ def run(self, path: str) -> Tuple[ort.InferenceSession, str]:
 
         model_as_string = model.SerializeToString()  # type: ignore
 
-        session = ort.InferenceSession(
-            model_as_string,
-            providers=[
-                "CPUExecutionProvider"
-                if os.environ["device"] == "cpu"
-                else "CUDAExecutionProvider"
-            ],
-        )
-
         basename = os.path.splitext(os.path.basename(path))[0]
 
-        return session, basename
+        return model_as_string, basename
+
+
+@NodeFactory.register("chainner:onnx:save_model")
+class OnnxSaveModelNode(NodeBase):
+    """ONNX Save Model node"""
+
+    def __init__(self):
+        super().__init__()
+        self.description = """Save ONNX model to file (.onnx)."""
+        self.inputs = [OnnxModelInput(), DirectoryInput(), TextInput("Model Name")]
+        self.outputs = []
+        self.category = ONNX
+        self.name = "Save Model"
+        self.icon = "ONNX"
+        self.sub = "Input & Output"
+
+        self.side_effects = True
+
+    def run(self, onnx_model: bytes, directory: str, model_name: str) -> None:
+        full_path = f"{os.path.join(directory, model_name)}.onnx"
+        with BytesIO() as f:
```

Comment:
Is this with statement necessary? It doesn't seem like you use f anywhere

Reply:
Whoops, good catch. Leftover code from a different approach.